pkg/crc/preflight/preflight_darwin.go

```diff
@@ -39,12 +39,42 @@ var dnsPreflightChecks = [...]PreflightCheck{
 	},
 }
 
-var traySetupCheck = PreflightCheck{
-	checkDescription: "Checking if the tray is installed and running",
-	check:            checkTrayExistsAndRunning,
-	fixDescription:   "Installing and setting up tray app",
-	fix:              fixTrayExistsAndRunning,
-	flags:            SetupOnly,
+var traySetupChecks = [...]PreflightCheck{
+	{
+		checkDescription: "Checking installed tray version",
+		check:            checkTrayVersion,
+		fixDescription:   "Installing and setting up tray app",
+		fix:              fixTrayVersion,
+		flags:            SetupOnly,
+	},
```

Comment:
`fixTrayVersion` expects the tray to be installed and running since it has a call to `RestartAgent`. On a system where the tray was never installed, I don't think this is going to work as expected. We may need a test checking if the tray binary has been unpacked and is available, then the check that the tray is installed and running, and then the version check.
Not sure how to best approach this, as I think some other corner cases are going to be odd :-/

Reply:
a defensive strategy is not bad

we can expect:

   * there is no tray
   * there is a tray
      * it is not running
      * it is running
         * you are having the right permissions to stop/start

Reply:
> fixTrayVersion expects the tray to be installed and running since it has a call to RestartAgent. On a system where the tray was never installed, I don't think this is going to work as expected.

No, it just expects the tray to be installed, it doesn't have to be running. Installed for the tray means two things -- the tray binary is present and the launchd plist file exists.

Reply:
I believe the following steps should handle both a new install and upgrade of the tray:
1. Check if tray binary is present
2. Check if tray/daemon plist is present (if not fix would create plist and load using `launchctl load`)
3. Check if tray version is matching (if not the fix would download/extract the tray and restart the tray)
4. Check if both tray/daemon running.

Reply:
> No, it just expects the tray to be installed, it doesn't have to be running.

But `checkTrayVersion/fixTrayVersion` is the very first check in `traySetupChecks`, at this point you don't know if the tray is installed or not.

Reply:
> I believe the following steps should handle both a new install and upgrade of the tray:

Seems fine.

Reply:
> But checkTrayVersion/fixTrayVersion is the very first check in traySetupChecks, at this point you don't know if the tray is installed or not.

Ahh, right my bad..

Reply:
@anjannath resolved or will be resolved in follow-up?

Reply:
@gbraad Its resolved, i made the changes

Reply:
I see you have `checkTrayBinaryPresent`